src/test/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/TestHelpers.kt

```diff
@@ -438,3 +438,22 @@ fun ISMTemplate.toJsonString(): String {
     val builder = XContentFactory.jsonBuilder()
     return this.toXContent(builder, ToXContent.EMPTY_PARAMS).string()
 }
+
+fun <T> wait(
```

Comment:
What is purpose of this compared to waitFor?
Can you add a description above explaining when/how to use this

Reply:
From what I saw, it looks like whereas `waitFor` will continue until a timeout or until the code block succeeds, this `wait` will continue until at least a timeout (even if the code block succeeded without exceptions).

I agree though, it would help to have a description if we plan on keeping this.

Reply:
So this is just continuously executing `block()` every 100 milliseconds until the specified timeout (defaults to 10 seconds). Is that the intention? Maybe will make more sense when I see it used in tests

Reply:
Where is this used @qreshi @bowenlan-amzn ? Took a look through tests and couldn't find it

Reply:
only [here](https://github.com/opendistro-for-elasticsearch/index-management/pull/280/files/b1bd49bc1d2fcc4949bd5d322ec404a4b7df481e?file-filters%5B%5D=.kt#diff-36b064552094de22c662e2f98fda3a57f1f62e8cd56d1bbf8e3e7b07d774c6d4R229)